core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -172,25 +172,14 @@ public DynamicFiltersStats getDynamicFilteringStats(QueryId queryId, Session ses
         int replicatedFilters = context.getReplicatedDynamicFilters().size();
         int totalDynamicFilters = context.getTotalDynamicFilters();
 
+        ConnectorSession connectorSession = session.toConnectorSession();
         List<DynamicFilterDomainStats> dynamicFilterDomainStats = context.getDynamicFilterSummaries().entrySet().stream()
                 .map(entry -> {
                     DynamicFilterId dynamicFilterId = entry.getKey();
-                    Domain domain = entry.getValue();
-                    // simplify for readability
-                    String simplifiedDomain = domain.simplify(1).toString(session.toConnectorSession());
-                    int rangeCount = domain.getValues().getValuesProcessor().transform(
-                            Ranges::getRangeCount,
-                            discreteValues -> 0,
-                            allOrNone -> 0);
-                    int discreteValuesCount = domain.getValues().getValuesProcessor().transform(
-                            ranges -> 0,
-                            DiscreteValues::getValuesCount,
-                            allOrNone -> 0);
                     return new DynamicFilterDomainStats(
                             dynamicFilterId,
-                            simplifiedDomain,
-                            rangeCount,
-                            discreteValuesCount,
+                            // use small limit for readability
+                            entry.getValue().toString(connectorSession, 1),
```

Comment:
limit=1 actually unlocks some special behavior in SortedRangeSet, it's a magic value.

Reply:
got rid of the `limit=1` special behaviour, using limit=2 here now